applications/Chat/evaluate/evaluator.py

```diff
@@ -47,28 +58,73 @@ def switch(metric):
             else:
                 raise ValueError(f"Unexpected metric")
 
+        answers_per_category = get_data_per_category(answers, list(self.params.keys()))
+        targets_per_category = get_data_per_category(targets, list(self.params.keys()))
+
         # automatic evaluation
         for category in self.params:
             category_metrics = self.params[category]["Metrics"]
-            targets_list = []
-            predicts_list = []
-            self.stats[category] = {}
-            
-            for dict in targets:
-                if dict["category"] == category:
-                    if(dict["target"]):
-                        targets_list.append(dict["target"])
-                    else:
-                        targets_list.append(dict["output"])
-
-            for dict in answers: 
-                if dict["category"] == category:
-                    predicts_list.append(dict["output"])
+            self.automatic_metric_stats[category] = {}
+
+            targets_list = [
+                target["target"] if target["target"] else target["output"] for target in targets_per_category[category]
+            ]
+            predicts_list = [answer["output"] for answer in answers_per_category[category]]
 
             for metric in category_metrics:
-                self.stats[category].update(switch(metric=metric))
+                self.automatic_metric_stats[category].update(switch(metric=metric))
+
+        # gpt35 evaluation
+        for category in self.params:
+            category_metrics = self.params[category]["GPT-3.5"]
+
+            prompt = self.gpt_evaluation_prompt.get(category, None)
+            if prompt is None:
+                print(f"No prompt for category {category}! Use prompt for category general now.")
+                prompt = self.gpt_evaluation_prompt["general"]
+
+            self.gpt35_evaluation_results[category] = gpt_evaluate.gpt35_evaluate(answers_per_category[category],
+                                                                                  prompt, category_metrics, category)
```

Comment:
Do we support GPT-4? It seems that all functions are designed for GPT-3.5?

Reply:
For evaluation, we now only support GPT-3.5. Evaluation using GPT-4 requires sampling which costs a lot and we also don't have GPT-4 key to test it.

Reply:
Sure. We can add it into TODO list.